pkg/provisioner/ironic/delete_test.go

```diff
@@ -225,3 +225,102 @@ func deleteTest(t *testing.T, detach bool) {
 		})
 	}
 }
+
+func TestDeleteWithPowerOff(t *testing.T) {
+
+	nodeUUID := "33ce8659-7400-4c68-9535-d10766f07a58"
+	maxPowerOffRetryCount := 3
+
+	cases := []struct {
+		name        string
+		ironic      *testserver.IronicMock
+		inspector   *testserver.InspectorMock
+		priorErrors int
+
+		expectedPowerStateInRequest string
+		expectedError               string
+	}{
+		{
+			name: "test-power-change-without-errors",
+			ironic: testserver.NewIronic(t).WithDefaultResponses().Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+					PowerState:     powerOn,
+				},
+			),
+			priorErrors:                 0,
+			expectedPowerStateInRequest: "power off",
+			expectedError:               "",
+		},
+		{
+			name: "test-power-change-with-errors",
+			ironic: testserver.NewIronic(t).WithDefaultResponses().Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+					PowerState:     powerOn,
+				},
+			),
+			priorErrors:                 maxPowerOffRetryCount - 1,
+			expectedPowerStateInRequest: "power off",
+			expectedError:               "",
+		},
+		{
+			name: "test-power-change-with-max-errors",
+			ironic: testserver.NewIronic(t).WithDefaultResponses().Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+					PowerState:     powerOn,
+				},
+			).Delete(nodeUUID),
+			priorErrors:                 maxPowerOffRetryCount + 1,
+			expectedPowerStateInRequest: "",
```

Comment:
There should be a problem with the value assigned to expectedPowerStateInRequest.
 According to the above code, the machine should be poweroff when priorityErrors > maxPowerOffRetryCount.

Reply:
Thanks!